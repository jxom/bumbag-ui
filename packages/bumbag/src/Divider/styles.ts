import { cssClass } from '../styled';
import { palette, space, theme } from '../utils';

export const Divider = (styleProps) => cssClass`
    border: 1px solid ${palette('white800', { dark: 'gray700' })(styleProps)};
    border-width: 0 0 1px 0;
    padding: 0;
    width: auto;
    height: 0;

    &[aria-orientation="vertical"] {
      border-width: 0 1px 0 0;
      width: 0;
      height: auto;

      & {
        ${theme(styleProps.themeKey, `styles.vertical`)(styleProps)};
      }
    }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
