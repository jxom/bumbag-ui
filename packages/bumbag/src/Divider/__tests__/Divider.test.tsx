import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Divider } from '../Divider';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Divider />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Divider borderColor="primary" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for vertical orientation', () => {
    const { container } = render(<Divider orientation="vertical" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Divider use="div" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Divider props', () => {
      const { result } = renderHook(() => Divider.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Divider>{(DividerProps) => <div {...DividerProps} />}</Divider>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Divider.base should render correctly', () => {
    const { container } = render(<Divider />, {
      // @ts-ignore
      theme: { Divider: { styles: { base: { borderColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Divider.vertical.base should render correctly', () => {
    const { container } = render(<Divider />, {
      // @ts-ignore
      theme: { Divider: { styles: { vertical: { borderColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Divider />, {
      // @ts-ignore
      theme: { Divider: { defaultProps: { className: 'test', borderColor: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
