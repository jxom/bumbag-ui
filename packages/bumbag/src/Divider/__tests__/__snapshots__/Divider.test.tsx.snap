// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border: 1px solid #e6e6eb;
  border-width: 0 0 1px 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-0[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  width: 0;
  height: auto;
}

<div
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Divider"
  role="separator"
/>
`;

exports[`composition hook should return with Divider props 1`] = `
Object {
  "aria-orientation": "horizontal",
  "children": undefined,
  "className": "css-1vdoif7 css-4lfl7m css-1yuzjtp fp-Box fp-Divider",
  "role": "separator",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border: 1px solid #e6e6eb;
  border-width: 0 0 1px 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-0[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  width: 0;
  height: auto;
}

<hr
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Divider"
  role="separator"
/>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border: 1px solid #e6e6eb;
  border-width: 0 0 1px 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-0[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  width: 0;
  height: auto;
}

.emotion-2 {
  border-color: #574feb !important;
}

<hr
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-Divider"
  role="separator"
/>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border: 1px solid #e6e6eb;
  border-width: 0 0 1px 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-0[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  width: 0;
  height: auto;
}

<hr
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Divider"
  role="separator"
/>
`;

exports[`props should render correctly for vertical orientation 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border: 1px solid #e6e6eb;
  border-width: 0 0 1px 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-0[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  width: 0;
  height: auto;
}

<hr
  aria-orientation="vertical"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Divider"
  role="separator"
/>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border: 1px solid #e6e6eb;
  border-width: 0 0 1px 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-0[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  width: 0;
  height: auto;
}

.emotion-2 {
  border-color: #574feb !important;
}

<hr
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Divider"
  role="separator"
/>
`;

exports[`theming Divider.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border: 1px solid #e6e6eb;
  border-width: 0 0 1px 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-0[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  width: 0;
  height: auto;
}

.emotion-0.emotion-0 {
  border-color: #ff0000;
}

<hr
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Divider"
  role="separator"
/>
`;

exports[`theming Divider.vertical.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border: 1px solid #e6e6eb;
  border-width: 0 0 1px 0;
  padding: 0;
  width: auto;
  height: 0;
}

.emotion-0[aria-orientation="vertical"] {
  border-width: 0 1px 0 0;
  width: 0;
  height: auto;
}

.emotion-0[aria-orientation="vertical"].emotion-0[aria-orientation="vertical"] {
  border-color: #ff0000;
}

<hr
  aria-orientation="horizontal"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Divider"
  role="separator"
/>
`;
